posixlib/src/main/resources/scala-native/pthread.c

```diff
@@ -20,51 +20,46 @@ size_t scalanative_size_of_pthread_mutexattr_t() {
     return sizeof(pthread_mutexattr_t);
 }
 
-int scalanative_pthread_cancel_asynchronous() {
-    return PTHREAD_CANCEL_ASYNCHRONOUS;
-}
+int scalanative_cancel_asynchronous() { return PTHREAD_CANCEL_ASYNCHRONOUS; }
 
-int scalanative_pthread_cancel_enable() { return PTHREAD_CANCEL_ENABLE; }
+int scalanative_cancel_enable() { return PTHREAD_CANCEL_ENABLE; }
 
-int scalanative_pthread_cancel_ered() { return PTHREAD_CANCEL_DEFERRED; }
+int scalanative_cancel_ered() { return PTHREAD_CANCEL_DEFERRED; }
 
-int scalanative_pthread_cancel_disable() { return PTHREAD_CANCEL_DISABLE; }
+int scalanative_cancel_disable() { return PTHREAD_CANCEL_DISABLE; }
 
-void *scalanative_pthread_canceled() { return PTHREAD_CANCELED; }
+void *scalanative_canceled() { return PTHREAD_CANCELED; }
 
-int scalanative_pthread_create_deteached() { return PTHREAD_CREATE_DETACHED; }
+int scalanative_create_deteached() { return PTHREAD_CREATE_DETACHED; }
 
-int scalanative_pthread_create_joinale() { return PTHREAD_CREATE_JOINABLE; }
+int scalanative_create_joinale() { return PTHREAD_CREATE_JOINABLE; }
 
-int scalanative_pthread_explicit_sched() { return PTHREAD_EXPLICIT_SCHED; }
+int scalanative_explicit_sched() { return PTHREAD_EXPLICIT_SCHED; }
 
-int scalanative_pthread_inherit_sched() { return PTHREAD_INHERIT_SCHED; }
+int scalanative_inherit_sched() { return PTHREAD_INHERIT_SCHED; }
 
-int scalanative_pthread_mutex_ault() { return PTHREAD_MUTEX_DEFAULT; }
+int scalanative_mutex_ault() { return PTHREAD_MUTEX_DEFAULT; }
 
-int scalanative_pthread_mutex_errorcheck() { return PTHREAD_MUTEX_ERRORCHECK; }
+int scalanative_mutex_errorcheck() { return PTHREAD_MUTEX_ERRORCHECK; }
 
-int scalanative_pthread_mutex_normal() { return PTHREAD_MUTEX_NORMAL; }
+int scalanative_mutex_normal() { return PTHREAD_MUTEX_NORMAL; }
 
-int scalanative_pthread_mutex_recursive() { return PTHREAD_MUTEX_RECURSIVE; }
+int scalanative_mutex_recursive() { return PTHREAD_MUTEX_RECURSIVE; }
 
-pthread_once_t scalanative_pthread_once_init() {
-    pthread_once_t once_block = PTHREAD_ONCE_INIT;
-    return once_block;
-}
+pthread_once_t scalanative_once_init() { return PTHREAD_ONCE_INIT; }
 
-int scalanative_pthread_prio_inherit() { return PTHREAD_PRIO_INHERIT; }
```

Comment:
Most of these like this one were correct. The C namespace is `PTHREAD_PRIO_INHERIT` so the Scala Native namespace is `scalanative_pthread_prio_inherit`. We are only prepending to the symbol. The exception is that we lowercase uppercase constants so the function in C is `c_like_function`. The reduces cognitive load for the reader as there is very little translation. Also, maintainers don't have to think things up.

Reply:
I made the corrections as you described.   

I believe the set of methods which _did_not_ have the `pthread_` were methods which
do not conform to the Posix spec.  The difference in name was probably to indicate that.

Anyway, I converted the names but left a comment block above the section, so that
adding the required `pthread_` does not lead to loss of the information that these are
not Posix.   

And of course, changing the names in the C file triggered the need to move the functions
so that the alphabetical order currently in the file is maintained.  Yes, at some point things
might get changed to declaration order. I think it _necessary_ that each iteration in the
evolution to perfection maintain _some_ internal consistency. Mainly because people
read in the now and may not be privy to future plans and those plans may never bear
fruit.

Reply:
I think spec order is good - easier to compare.